enumerable_methods.rb

```diff
@@ -0,0 +1,145 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < size
+      yield self[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    selected_array = []
+    my_each do |element|
+      selected_array << element if yield(element)
+    end
+    selected_array
+  end
+
+  def my_all?(_para = nil)
+    result = true
+    if block_given?
+      my_each do |element|
+        result = false unless yield(element)
+      end
+    else
+      my_each do |element|
+        result = false unless element
+      end
+    end
+    result
+  end
+
+  def my_any?(para = nil)
+    return my_any?(para) if block_given? && !para.nil?
+
+    if block_given?
+      to_a.my_each do |element|
+        return true if yield element
+
+        false
+      end
+    elsif para
+      to_a.my_each do |element|
+        return true if element == para
+      end
+    elsif para.nil?
+      to_a.my_each do |element|
+        return true if element
+      end
+    end
+    false
+  end
+
+  def my_none?(para = nil)
+    return my_none?(para) if block_given? && !para.nil?
+
+    if block_given?
+      to_a.my_each do |element|
+        return false if yield element
+
+        false
+      end
+    elsif para
+      to_a.my_each do |element|
+        return false if element == para
+      end
+    elsif para.nil?
+      to_a.my_each do |element|
+        return false if element
+      end
+    end
+    true
+  end
```

Comment:
- [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_none?

- [ ] when a class is passed as an argument returns true if none of the collection is a member of such class
```ruby
[1, 2.5, 'a', 9].my_none?(Integer) == [1, 2.5, 'a', 9].none?(Integer) => false
```
- Yours is retrieving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/80104949-217fb580-8535-11ea-9dde-ac1543513af2.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690539-130ba180-6f69-11ea-8bd0-6ca061ab05d2.png)

- [ ] when a Regex is passed as an argument returns true only if none of the collection matches the Regex
```ruby
%w[dog door rod blade].my_none?(/o/) == %w[dog door rod blade].none?(/o/) => false
```
- Yours is retrieving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/80105072-4b38dc80-8535-11ea-899d-4b5bac91d27e.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/80105114-57bd3500-8535-11ea-8341-4b2d78b39bef.png)

[OPTIONAL]

I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:

  def my_none?(arg = nil, &block)
    !my_any?(arg, &block)
  end

To make this work we have to use block.call(element) instead of yield(element) in my_any?